module-network.tf

```diff
@@ -0,0 +1,142 @@
+
+data "oci_core_vcn" "oke" {
+  count  = var.create_vcn ? 0 : 1
+  vcn_id = var.vcn_id
+}
+
+locals {
+  # Created VCN if enabled, else var.vcn_id
+  vcn_id = var.create_vcn ? one(module.vcn[*].vcn_id) : var.vcn_id
+
+  # Configured VCN CIDRs if creating, else from provided vcn_id
+  vcn_cidrs = var.create_vcn ? var.vcn_cidrs : flatten(data.oci_core_vcn.oke[*].cidr_blocks)
+
+  # Created route table if enabled, else var.ig_route_table_id
+  ig_route_table_id = var.create_vcn ? one(module.vcn[*].ig_route_id) : var.ig_route_table_id
+
+  # Created route table if enabled, else var.nat_route_table_id
+  nat_route_table_id = var.create_vcn ? one(module.vcn[*].nat_route_id) : var.nat_route_table_id
+}
+
+module "vcn" {
+  count          = var.create_vcn ? 1 : 0
+  source         = "oracle-terraform-modules/vcn/oci"
+  version        = "3.5.3"
+  compartment_id = coalesce(var.network_compartment_id, local.compartment_id)
+  defined_tags   = var.defined_tags["vcn"]
+  freeform_tags  = var.freeform_tags["vcn"]
+
+  attached_drg_id              = var.drg_id != null ? var.drg_id : (var.create_drg ? module.drg[0].drg_id : null)
+  create_internet_gateway      = !(var.load_balancers == "internal" && !var.create_bastion && var.control_plane_type == "private")
+  create_nat_gateway           = var.worker_type == "private" || var.create_operator || var.load_balancers == "internal" || var.load_balancers == "both"
+  create_service_gateway       = true
+  internet_gateway_route_rules = var.internet_gateway_route_rules
+  local_peering_gateways       = var.local_peering_gateways
+  lockdown_default_seclist     = var.lockdown_default_seclist
+  nat_gateway_public_ip_id     = var.nat_gateway_public_ip_id
+  nat_gateway_route_rules      = var.nat_gateway_route_rules
+  vcn_cidrs                    = local.vcn_cidrs
+  vcn_dns_label                = var.assign_dns ? var.vcn_dns_label : null
+  vcn_name                     = var.vcn_name
+}
+
+module "drg" {
+  count          = var.create_drg || var.drg_id != null ? 1 : 0
+  source         = "oracle-terraform-modules/drg/oci"
+  version        = "1.0.3"
+  compartment_id = coalesce(var.network_compartment_id, local.compartment_id)
+
+  drg_id           = var.drg_id # existing DRG ID or null
+  drg_display_name = var.drg_display_name
+  drg_vcn_attachments = {
+    (random_id.state_id.id) : {
+      vcn_id                    = local.vcn_id
+      vcn_transit_routing_rt_id = null
+      drg_route_table_id        = null
+    }
+  }
+}
+
+module "network" {
+  source           = "./modules/network"
+  state_id         = random_id.state_id.id
+  compartment_id   = coalesce(var.network_compartment_id, local.compartment_id)
+  defined_tags     = lookup(var.defined_tags, "network", {})
+  freeform_tags    = lookup(var.freeform_tags, "network", {})
+  tag_namespace    = var.tag_namespace
+  use_defined_tags = var.use_defined_tags
+
+  allow_node_port_access       = var.allow_node_port_access
```

Comment:
For this set of access rules, they are either ingress or egress. Should we indicate these in the names?

Reply:
Possibly, though others e.g. `allow_worker_ssh_access` for worker ingress also include creation of bastion egress. Let's consider in another change separate from file organization.